### PR TITLE
[tensorflow-sys] Expose the eager API.

### PR DESCRIPTION
**Pull request by dskkato**
In this PR, the eager APIs are exposed via `tensorflow-sys`. The eager API is a very experimental stage, so it is separated as `eager` features.

The binding generation is done by using TF2.6. The original binding code is renamed from `bindgen.rs` to `c_api.rs`, and non-TF_* related symbols were removed.

This PR is related to #321 and #326.

### CODE REVIEW

File: tensorflow-sys/tests/lib.rs [hunk 1]
```diff
@@ -8,3 +9,32 @@ fn linkage() {
...
     }
 }
+
+/// Test that the eager API works.
+#[cfg(feature = "eager")]
+#[test]
+fn tfe_tensor_handle() {
+    let data = vec![0.0f32; 100];
+    let shape = [1, 10, 10];
+    let num_elements = shape.iter().fold(1, |t, x| t * x) as usize;
```
Comment (adamcrume) [inline tensorflow-sys/tests/lib.rs:19]: This can be simplified to `shape.iter().product()`.

File: tensorflow-sys/tests/lib.rs [hunk 2]
```diff
@@ -8,3 +9,32 @@ fn linkage() {
...
+            num_elements * std::mem::size_of::<f32>(),
+        );
+        std::ptr::copy_nonoverlapping(
+            data.as_ptr(),
+            ffi::TF_TensorData(tf_tensor) as *mut f32,
+            ffi::TF_TensorByteSize(tf_tensor) / std::mem::size_of::<f32>(),
+        );
+        let status = ffi::TF_NewStatus();
+        let tfe_handle = ffi::TFE_NewTensorHandle(tf_tensor, status);
+
```
Comment (adamcrume) [inline tensorflow-sys/tests/lib.rs:35]: This should check that the status is OK.

**Review by adamcrume — CHANGES_REQUESTED** [review]

File: tensorflow-sys/tests/lib.rs (hunk 1 above)
Comment (dskkato) [reply tensorflow-sys/tests/lib.rs]: Thanks. The code now uses `product`.

File: tensorflow-sys/tests/lib.rs (hunk 2 above)
Comment (dskkato) [reply tensorflow-sys/tests/lib.rs:43]: The status check code is a copy of the `ok!` macro in the `multiplication.rs`.

**Review by adamcrume — APPROVED** [review]

**Comment by adamcrume** [comment]
Thanks!

**Pull request merged**